### PR TITLE
Let GenerateCss also generate .min.css

### PR DESCRIPTION
**Pull request by dgrammatiko**
This is a response to #6378
#### What it does?

It adds a minified version for every stylesheet for templates isis and protostar
#### Testing

Apply patch
Go to front end (protostar)
And admin area (isis)
Also try to enable debug
#### B/C

This should be 100% backwards compatible

### CODE REVIEW

File: administrator/templates/isis/component.php [hunk 1]
```diff
@@ -20,8 +20,15 @@
 
 $doc->addScript($this->baseurl . '/templates/' . $this->template . '/js/template.js');
 
-// Add Stylesheets
-$doc->addStyleSheet($this->baseurl . '/templates/' . $this->template . '/css/template.css');
+// Load template CSS
+if (!$app->get('debug_lang', '0') == '1' || !$app->get('debug', '0') == '1')
```
Comment (mbabker) [inline administrator/templates/isis/component.php]: Just check `if (JDEBUG)`... That's what the constant is there for.

Comment (dgt41) [inline administrator/templates/isis/component.php]: Done! Thanks @mbabker

File: administrator/templates/isis/index.php [hunk 2]
```diff
@@ -22,7 +22,15 @@
...
-$doc->addStyleSheetVersion($this->baseurl . '/templates/' . $this->template . '/css/template' . ($this->direction == 'rtl' ? '-rtl' : '') . '.css');
+$css_path = $this->baseurl . '/templates/' . $this->template . '/css/template' . ($this->direction == 'rtl' ? '-rtl' : '') . '.min.css';
+
+if (JDEBUG)
+{
+	$css_path = $this->baseurl . '/templates/' . $this->template .
+		'/css/template' . ($this->direction == 'rtl' ? '-rtl' : '') . '.css';
+}
+
+	$doc->addStyleSheetVersion($css_path);
```
Comment (Hackwar) [inline administrator/templates/isis/index.php]: You can replace all this (almost) with 
```php
JHtml::_('stylesheet', 'template' . ($this->direction == 'rtl' ? '-rtl' : '') . '.css', array(), true);
```
It wont use $doc->addStyleSheetVersion(), which we maybe should fix.

File: administrator/templates/isis/css/template-rtl.css [hunk 3]
```diff
@@ -8047,13 +8047,13 @@ a.grid_true {
 	padding: 0;
 }
 .j-toggle-button-wrapper.j-toggle-hidden {
-	right: -20px;
+	right: -24px;
```
Comment (Hackwar) [inline administrator/templates/isis/css/template-rtl.css]: Why are you changing these?

File: templates/protostar/error.php [hunk 4]
```diff
@@ -51,6 +51,15 @@
...
 }
+
+// Add Stylesheets
+$css_path = $this->baseurl . '/templates/' . $this->template . '/css/template.min.css';
+
+if (JDEBUG)
+{
+	$css_path = $this->baseurl . '/templates/' . $this->template . '/css/template.css';
+}
+
```
Comment (Hackwar) [inline templates/protostar/error.php]: I was about to point to my above statement about JHtml, but just as a reminder for others looking at this: We want to keep the coupling to other parts of the system as minimal as possible for an error page. So this is cool. :wink:

File: administrator/templates/isis/css/template-rtl.css (hunk 3 above)
Comment (dgt41) [inline administrator/templates/isis/css/template-rtl.css]: I have no idea why this did happen! Honestly I didn't touch any of the less files! Somebody should have made changes without using generatecss.php.

File: administrator/templates/isis/index.php (hunk 2 above)
Comment (dgt41) [inline administrator/templates/isis/index.php]: Versioning, at least for admin area, is essential. So I wouldn't use  `JHtml` yet, but I do like the suggestion to bring versioning in that method :+1:

**Comment by nonumber** [comment]
I was also doing this and started with this PR: https://github.com/joomla/joomla-cms/pull/6383

File: administrator/templates/isis/css/template-rtl.css (hunk 3 above)
Comment (JoomliC) [inline administrator/templates/isis/css/template-rtl.css]: This is a change i've done in another PR merged : https://github.com/joomla/joomla-cms/pull/6327
But don't know why it appear in this PR...
